src/test/java/StringManipulationTest.java

```diff
@@ -49,101 +55,137 @@ public void testRemoveNthCharacter1() {
 
     @Test
     public void testRemoveNthCharacter2() {
-        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        manipulatedstring.setString("Hi this is the message");
+        assertEquals("H   h s i   h   e s g ", manipulatedstring.removeNthCharacter(2, true));
     }
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This message should have less character deleted");
+        assertEquals("This me sage sh uld hav  less c aracter deleted", manipulatedstring.removeNthCharacter(8, true));
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This message should have only the last character deleted");
+        assertEquals("This message should have only the last character delete", manipulatedstring.removeNthCharacter(56, false));
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This message should be unreadable");
```

Comment:
I'd recommend adding a test that checks to see what happens when N is a value larger than the length of the string

Reply:
yes I did that now